eslint/babel-eslint-parser/src/analyze-scope.js

```diff
@@ -166,6 +166,10 @@ class Referencer extends OriginalReferencer {
     this._visitClassProperty(node);
   }
 
+  ClassPrivateMethod(node) {
+    super.MethodDefinition(node);
```

Comment:
May worth a todo item here that once TypeScript or Flow supports private method, `ClassPrivateMethod` should visit `typeAnnotations` before `super.MethodDefinition`, of which the similar has been done in `_visitClassProperty`.